test/unit/views/ArrivalDetailsViewSpec.scala

```diff
@@ -62,7 +66,12 @@ class ArrivalDetailsViewSpec extends ViewSpec with Injector {
         val backButton = emptyView.getBackButton
 
         backButton mustBe defined
-        backButton.get must haveHref(controllers.routes.ConsignmentReferencesController.displayPage())
+        backButton.get must haveHref(
+          if (realAppConfig.ileQueryEnabled)
+            controllers.routes.ChoiceController.displayChoiceForm()
+          else
+            controllers.routes.ConsignmentReferencesController.displayPage()
+        )
```

Comment:
IMO all cases should be tested every time on unit level. So instead of using real app config, I'd advise to use a mock and have 2 test cases depending on whether the feature flag is enabled or not.

Reply:
I was wondering about that when I did that (not in this PR).
Really, the href doesn't need to be tested here because it's tested in the config helper class that generates the url.  It should be enough to test that there _is_ a back link and that's all.

Reply:
In that case I'd do exactly as you said: "...to test that there is a back link and that's all", but without checking the urls. If it was like that, then I'd expect that the url itself is tested somewhere else. But ATM the intention is not clear to me as a reader.